doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -82,7 +81,8 @@ associativity rules have to be given.
    The right-hand side of a notation is interpreted at the time the notation is
    given. In particular, disambiguation of constants, :ref:`implicit arguments
    <ImplicitArguments>` and other notations are resolved at the
-   time of the declaration of the notation.
+   time of the declaration of the notation. The right-hand side is
+   currently typed only at use time but this may change in the future.
```

Comment:
>this may change in the future

For what purpose?

Reply:
Because I find it a bit counterintuitive to have an interpretation dependent on dynamic elaboration state. And also that errors are reported at use time.

Or is the question about the purpose of alerting? If so, it is to protect myself, but maybe that worries more than needed?